x-pack/legacy/plugins/lens/public/editor_frame_plugin/merge_tables.ts

```diff
@@ -38,16 +42,35 @@ export const mergeTables: ExpressionFunction<
     },
   },
   context: {
-    types: ['null'],
+    types: ['kibana_context', 'null'],
   },
-  fn(_ctx, { layerIds, tables }: MergeTables) {
+  fn(ctx, { layerIds, tables }: MergeTables) {
     const resultTables: Record<string, KibanaDatatable> = {};
     tables.forEach((table, index) => {
       resultTables[layerIds[index]] = table;
     });
     return {
       type: 'lens_multitable',
       tables: resultTables,
+      dateRange: getDateRange(ctx),
     };
   },
 };
+
+function getDateRange(ctx?: KibanaContext | null) {
+  if (!ctx || !ctx.timeRange) {
+    return;
+  }
+
+  const fromDate = dateMath.parse(ctx.timeRange.from);
+  const toDate = dateMath.parse(ctx.timeRange.to);
```

Comment:
There might be a time zone parameter on these, can you check?

Reply:
Doesn't look like it.